src/access/for.ts

```diff
@@ -177,3 +180,131 @@ export async function getAccessForAll(
+ * Set access to a Resource for everyone.
+ *
+ * This function works with Solid Pods that implement either the Web Access
+ * Control spec or the Access Control Policies proposal, with some caveats:
+ *
+ * - If access to the given Resource has been set using anything other than the
+ *   functions in this module, it is possible that it has been set in a way that
+ *   prevents this function from reliably setting access, in which case it will
+ *   resolve to `null`.
+ * - It will only set access explicitly for the general public.
+ *   In other words, additional restrictions could be present that further
+ *   restrict or loosen what access everyone has in particular
+ *   circumstances.
+ * - The provided access will only apply to the given Resource. In other words,
+ *   if the Resource is a Container, the configured Access may not apply to
+ *   contained Resources.
+ * - If the current user does not have permission to view or change access for
+ *   the given Resource, this function will resolve to `null`.
+ *
+ * Additionally, two caveats apply to users with a Pod server that uses WAC:
+ * - If the Resource did not have an ACL yet, a new one will be initialised.
+ *   This means that changes to the ACL of a parent Container can no longer
+ *   affect access people have to this Resource, although existing access will
+ *   be preserved.
+ * - Setting different values for `controlRead` and `controlWrite` is not
+ *   supported, and **will throw an error**. If you expect (some of) your users
+ *   to have Pods implementing WAC, be sure to pass the same value for both.
+ *
+ * @param resourceUrl URL of the Resource you want to change the Agent's access to.
+ * @param actorType type of actor whose access is being read.
+ * @param access What access permissions you want to set for the given Agent to the given Resource. Possible properties are `read`, `append`, `write`, `controlRead` and `controlWrite`: set to `true` to allow, to `false` to stop allowing, or `undefined` to leave unchanged. Take note that `controlRead` and `controlWrite` can not have distinct values for a Pod server implementing Web Access Control; trying this will throw an error.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns What access has been set for the given Agent explicitly.
+ */
+export async function setAccessFor(
+  resourceUrl: UrlString,
+  actorType: "public",
+  access: Partial<Access>,
+  options?: typeof internal_defaultFetchOptions
+): Promise<Access | null>;
+export async function setAccessFor(
+  resourceUrl: UrlString,
+  actorType: Actor,
+  access: Partial<Access>,
+  actor?: WebId | UrlString | typeof internal_defaultFetchOptions,
+  options?: typeof internal_defaultFetchOptions
+): Promise<Access | null>;
+export async function setAccessFor(
```

Comment:
When an overload is specified, the function's own type is hidden. Since some libraries might not yet know the type of actor (i.e. they're just passing along an arbitrary string passed to them), it can be annoying if only the specific overloads are available. To avoid annoying warnings there, we can add the generic type as a fallback overload:

```suggestion
export async function setAccessFor(
  resourceUrl: UrlString,
  actorType: Actor,
  access: Partial<Access>,
  actor?:
    | WebId
    | UrlString
    | typeof internal_defaultFetchOptions,
  options?: typeof internal_defaultFetchOptions
): Promise<Access | null>;
export async function setAccessFor(
```

Reply:
I'm a bit confused: isn't this already the case ? If I understand this correctly, there are three overloads currently specified, the two first (with the docblocks) being specific, the third one being generic, and then the actual function is declared ? Anyways, now I understand better why the last overload matches the function signature ^^.

Reply:
Ehm, well, you know, that might be the case if you actually look properly, but if, like me, you just don't look properly, then you can easily pretend that the overload does not exist.

Jokes aside, you're right, I didn't look properly. Carry on!